neet/boolean/sensitivity.py

```diff
@@ -479,8 +490,243 @@ def average_sensitivity(self, states=None, weights=None, calc_trans=True):
+
+        if states is not None:
+            num_states = 0
+            states = list(states)
+            #print("type of states: ", type(states))
+            #print("len of states: ", type(states))
+            #print("len of states: ", len(states))
+            #print("len of states: ", len(states))
+            if calc_trans:
+                decoder = self.decode
+                trans = list(map(decoder, self.transitions))
+            else:
+                trans = None
+
+        """
+
+
+        s = 0
+
+        # optionally pre-calculate transitions
+        if calc_trans:
+            decoder = self.decode
+            trans = list(map(decoder, self.transitions))
+        else:
+            trans = None
+
+
+        if states is not None:
+            # Iterate through all provided states and sum 
+            # the distances between them and their c-hamming-neighbors
+            # after a single time-step (one synchronous transition)
+            for state in states:
+                s += self.c_sensitivity(state, trans, c)
+
+            # get the average of s by diving the sum by the
+            # number of states considered
+            s = s / len(states)
+
+
+        else:
+            # Generate all possible states 
+            # and sum the distances between them and each of their
+            # c-hamming-neighbors in the next time step (after one 
+            # synchronous transition)
+            for state in self:
+                s += self.c_sensitivity(state, trans, c)
+
+
+            # Deprecated implementation I'm keeping around while we implement unit testing. Will be removed after
+            # equivalence/correctness of the above (2 line) implementation is established
+            """for n in range(self.size):
+                state_gen = itt.combinations(range(self.size),n)
+                for state in state_gen:
+                    state_array = [0 for x in range(self.size)]
+                    for index in state:
+                        state_array[index] = 1
+                    s += self.c_sensitivity(state_array, trans, c)"""
+
+            # Average s by diving the sum by the 
+            # total number of possible states (2^n)
+            # s = s / np.power(2, self.size)
+            s = s / self.volume
+            """ 
+            s is now the average C-Sensitivity of f and must lie in the interval [0, (n choose c)] 
+            where n is the size of the network.
+            """
+
+            upper_bound = math.factorial(self.size) / (math.factorial(c) * math.factorial(self.size - c))
+            if s > upper_bound or s < 0:
+                raise RuntimeError('This value of S should not be possible and the code is therefore wrong')
+
+        return s
+        #yield s / upper_bound # yields the normalized average c-sensitivity
+
+    def derrida_plot(self, max_c=None):#, transitions=None): #X-Axis = c value, Y-Axis = output of Average_c_sensitivity
```

Comment:
So, it seems this plot isn't actually plotting the data! You'll need to call `plt.plot` with `x` and `y` coordinates, and it might be good to add some default labels on the axes.

I'm thinking something like...
```python
f, x = plt.subplots() # this will create a figure `f` and an axes object `ax`
ax.set_title('Derrida Plot') # set the title for this particular plot
ax.plot(range(max_c), yvals) # plot y-values vs. x-values
ax.set_xlabel('perturbation size (c)') # set the x-axis label
ax.set_ylabel('sensitivity') # the y-axis label
return f, ax # return the figure and the axes object.
```
By returning the figure and axes, end users can further modify the plots before they save them.